test/test_core.py

```diff
@@ -9494,9 +9494,13 @@ def test_embind_lib_with_asyncify(self, args):
 
   @no_asan('asyncify stack operations confuse asan')
   @no_wasm64('TODO: asyncify for wasm64')
+  @with_asyncify_and_stack_switching
   def test_em_async_js(self):
     self.uses_es6 = True
-    self.set_setting('ASYNCIFY')
+    if not self.get_setting('ASYNCIFY'):
+      self.set_setting('ASYNCIFY')
+    else:
+      self.set_setting('EXIT_RUNTIME')
```

Comment:
Is EXIT_RUNTIME always needed here?  Why?

Also, why not just unconditionally set both of these here?

Reply:
`with_asyncify_and_stack_switching` sets ASYNCIFY=2, it was getting overridden by doing `set_setting('ASYNCIFY')`. Also, ASYNCIFY==2 requires EXIT_RUNTIME. 
I suppose it'd be more clear if I did something like
```
if self.get_setting('ASYNCIFY') == 2:
  self.set_setting('EXIT_RUNTIME')
else:
  self.set_setting('ASYNCIFY')
```

Reply:
I did a bit of digging and the `EXIT_RUNTIME` looks to be a requirement of the old stack switching code. Seems fine without it now.